### PR TITLE
DATAUP-377 - Merge develop branch into truss branch

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This is an ongoing attempt to keep the truss and develop branches of work more or less in sync and work out issues as we go. Also, all of the dependency version bumps and production bug fixes are in the develop branch and need to get moved into the truss branch.

Tagging this with the DATAUP-377 ticket, as it's related work, though not in the current sprint scope.

I'm using an intermediate branch here to resolve conflicts. There weren't many, and the ugliest were in the dependency files `package.json` and `requirements.txt`.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-377
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [ ] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

No actual code changes that haven't been vetted elsewhere, so this checklist is pretty much moot.

- [ ] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [ ] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)
n/a
- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePublicTab.js [hunk 1]
```diff
@@ -412,7 +405,11 @@ define([
 
             this.resultPanel = $('<div role="table" data-test-id="result">');
 
-            this.resultsFooterMessage = $('<div>');
+            this.resultsFooterMessage = $('<div>')
+                .css('display', 'flex')
+                .css('flex-direction', 'row')
+                .css('align-items', 'center')
+                .css('justify-content', 'center');
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePublicTab.js:412]: how did this inline css sneak in? 😞

File: kbase-extension/static/kbase/js/widgets/narrative_core/publicDataSources/search2DataSource.js [hunk 2]
```diff
@@ -80,13 +81,46 @@ define(['bluebird', 'handlebars', 'common/searchAPI2', './common'], function (
                 track_total_hits: true,
             };
             if (query !== null) {
-                params.query.bool.must.push({
-                    match: {
-                        scientific_name: {
-                            query: query,
+                // Note that this rather convoluted-looking filter how one implements a required "or" filter.
+                // Having the "should" within a "must" ensures that the at least one of the "should" matches
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/publicDataSources/search2DataSource.js:85]: some odd grammar here!

**Review by ialarmedalien — APPROVED** [review]
👍

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePublicTab.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePublicTab.js:412]: That was just reformatted from how it was before. I think it'll have to be part of a greater effort...

**Pull request merged**